modulegen/internal/module/main.go

```diff
@@ -1,13 +1,53 @@
 package module
 
 import (
+	"fmt"
 	"path/filepath"
 	"strings"
 	"text/template"
 
+	"github.com/testcontainers/testcontainers-go/modulegen/internal/mkdocs"
+	"github.com/testcontainers/testcontainers-go/modulegen/internal/model"
+	"github.com/testcontainers/testcontainers-go/modulegen/internal/modfile"
 	internal_template "github.com/testcontainers/testcontainers-go/modulegen/internal/template"
 )
 
+func GenerateGoModule(ctx *model.Context, example model.Example) error {
```

Comment:
At some point we will be writing our own `FileGenerator` interface, with the `Generate() error` function 😅

Reply:
That can indeed be an idea.